ui/src/components/TimePicker/__tests__/Time.test.ts

```diff
@@ -0,0 +1,177 @@
+import { mount } from "@vue/test-utils";
+import Time from "../Time/Time.vue";
+
+describe("Time", () => {
+  let value = "23";
+  const placeholder = "Hour";
+  const ariaLabel = "24-Hour";
+
+  const wrapper = mount(Time, {
+    propsData: {
+      type: "hour24",
+      placeholder,
+      ariaLabel,
+      value
+    }
+  });
+
+  const input = wrapper.find("input");
+  expect(input).toBeDefined();
+
+  it("renders correctly", () => {
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it("renders correct placeholder and aria-label", () => {
+    expect(input.attributes("placeholder")).toBe(placeholder);
+    expect(wrapper.find(".fd-time__item").attributes("aria-label")).toBe(
+      ariaLabel
+    );
+  });
+
+  it("emits events and has correct inputValue and input element value", () => {
+    const navDown = wrapper.find(".sap-icon--navigation-down-arrow");
+    navDown.trigger("click");
+
+    const expected = "22";
+
+    expect(wrapper.emitted("input").length).toBeGreaterThan(0);
+
+    const timeUpdate = wrapper.emitted("timeUpdate");
+    expect(timeUpdate.length).toBeGreaterThan(0);
+    expect(timeUpdate[timeUpdate.length - 1][0]).toEqual(expected);
+
+    expect(wrapper.vm.$data.inputValue).toEqual(expected);
+    expect((input.element as HTMLInputElement).value).toEqual(expected);
+  });
+
+  describe("Hour24", () => {
+    const wrapper = mount(Time, {
+      propsData: {
+        type: "hour24",
+        value: "23"
+      }
+    });
+
+    it("navigates to correct value", () => {
+      const navUp = wrapper.find(".sap-icon--navigation-up-arrow");
+      const navDown = wrapper.find(".sap-icon--navigation-down-arrow");
+
+      navUp.trigger("click");
+      expect(wrapper.vm.$data.inputValue).toEqual("00");
```

Comment:
Why are you using `$data` here and not just `wrapper.vm.inputValue`? I approve - just curious.

Reply:
Hmm.. I think because typescript is making noise. I didn't define `inputValue` as data during mounting.

Reply:
I see.